src/hud.cpp

```diff
@@ -80,6 +81,43 @@ Hud::Hud(video::IVideoDriver *driver, scene::ISceneManager* smgr,
 	use_hotbar_image = false;
 	hotbar_selected_image = "";
 	use_hotbar_selected_image = false;
+
+	m_selectionmesh = NULL;
+	m_selectionboxes.clear();
+	m_highlighted_pos = v3f(0.0, 0.0, 0.0);
+	std::string mode = g_settings->get("node_highlighting");
+	m_selection_material.Lighting = false;
+	if (g_settings->getBool("enable_shaders")) {
+		IShaderSource *shdrsrc = gamedef->getShaderSource();
+		u16 sid;
+		if (mode == "halo") {
+			sid = shdrsrc->getShader("selection_shader", 1, 1);
```

Comment:
This would really be cleaner as a ternary function.  There's too much repetitive code otherwise.